### PR TITLE
Cleanup the codebase, add documentation, explanations, etc... 

### PR DESCRIPTION
**Pull request by AiroPi**
This pull request updates and upgrades the entire project from both a documentation and codebase perspective. Its aim is to enhance the project's contribution-friendliness by removing unnecessary elements, renaming components in a more convenient manner, and adding extensive general explanations.

Here's a non-exhaustive list of changes:

- Removed the "beta"-tagged compose file. This can now be manually adjusted during bot deployment, eliminating the need for a separate file.
- Replaced pylint (which wasn't enabled anyway), isort, bandit, and black with ruff. Ruff offers significant speed improvements and additional functionalities.
- Ensured compliance with new rules specified in pyproject.toml throughout the entire codebase.
- Implemented pip-tools to streamline dependency management.
- Updated versions of tools used in Github Actions.
- Added useful information in CONTRIBUTING.md
- And more...

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,33 +1,61 @@
...
+## Docker watch
+
+The [compose file](/compose.yml) implements [watch](https://docs.docker.com/compose/file-watch/) to help debugging the code. By running `docker compose watch`, the bot will be executed while observing for files changes or deps updates. This will allow extensions reload, and speed up restarts (rebuild not needed).
+
+## Run the code on debug version
+
+You can use [debugpy](https://github.com/microsoft/debugpy) easily when running MyBot on local, for dev and debug purposes.  
+Replace `docker compose` by `docker compose -f compose.yml -f compose.debug.yml` when using Docker Compose. Then you can use debugpy within the port `5678`.
+
+Additionally, this will setup the `config.DEBUG` to `True` from the code pov, it will set the logger level to DEBUG, and expose the PostgresSQL port to the host (`5432`).
```
Comment (nc0fr) [inline .github/CONTRIBUTING.md:39]: Stay consistent and write DEBUG within inline code block quotes (`DEBUG`).

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,33 +1,61 @@
...
 
 ## Translations contributions
 
 MyBot is a multi-language bot! The codebase is in english, which is then translated in several languages.
```
Comment (nc0fr) [inline .github/CONTRIBUTING.md:43]: English, like many other languages, starts with an uppercase letter (unlike French).

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -1,33 +1,61 @@
...
 
 Currently, MyBot is translated in :
-  -  French
+
+- French
 
 If you know one of these languages, you can contribute to translations here:
 https://crowdin.com/project/mybot-discord
 
 If you are able to add a new language to the bot, please contact me on Discord! It would be a pleasure to add a new language to MyBot!
```
Comment (nc0fr) [inline .github/CONTRIBUTING.md:52]: I feel like "If you want to add" would sounds friendlier than "If you are able to add." But I may just be nitpicking here.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -1,33 +1,61 @@
...
 However, this language must also be available on the Discord application.
 
-## Financial contributions 
+## Financial contributions
 
 You can make voluntary donation at https://www.buymeacoffee.com/airopi
```
Comment (nc0fr) [inline .github/CONTRIBUTING.md:57]: Donation should be in plural form.

**Review by nc0fr — CHANGES_REQUESTED** [review]
Great initiative! The document makes wanting to contribute to the codebase way more attracting. My feedback should more be seen as a way of improving and not a way to dismiss your goals, as everything looks great otherwise.

**Comment by AiroPi** [comment]
Some work-in-progress information got added by mistake, it will be completed later.

**Comment by AiroPi** [comment]
@nc0fr btw, do not care about the 60+ edited files. Just review the markdown files when they are updated. Thank you!

**Comment by nc0fr** [comment]
😭

**Review by nc0fr — APPROVED** [review]
LGTM

**Comment by nc0fr** [comment]
Ship it to production 🚀🚀

**Pull request merged**